semver.py

```diff
@@ -47,10 +27,69 @@ def cmp(a, b):
         return (a > b) - (a < b)
 
 
+def deprecated(func=None, replace=None, version=None, category=DeprecationWarning):
+    """
+    Decorates a function to output a deprecation warning.
+
+    This function will be removed once major version 4 of semver is
+    released.
+
+    :param str replace: the function to replace (use the full qualified
+        name like ``semver.VersionInfo.bump_major``.
+    :param str version: the first version when this function was deprecated.
+    :param category: allow you to specify the deprecation warning class
+        of your choice. By default, it's  :class:`DeprecationWarning`, but
+        you can choose :class:`PendingDeprecationWarning``or a custom class.
+    """
+
+    def decorator():
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            msg = ["Function 'semver.{f}' is deprecated."]
+
+            if version:
+                msg.append("Deprecated since version {v}.")
+
+            if replace:
+                msg.append("Use {r!r} instead.")
+            else:
+                msg.append("Use the respective 'semver.VersionInfo.{r}' instead.")
+
+            # warnings.simplefilter('always', category)
+            frame = inspect.currentframe().f_back
```

Comment:
May I ask what happens here?

Reply:
Of course! This is needed to get the frame object which contains the line number, the filename etc. from the originated deprecated function. This is then passed to the `warnings.warn_explict` around line 63.

Reply:
Ah, I see, thanks :)